### DISCUSSION TITLE
Is it possible to cancel a running check-sat request?

### DISCUSSION THREAD

**Discussion by keyboardDrummer** [Z3Prover/z3#5837] [Q&A] [answered]
After sending a `(check-sat)` command to Z3, is it possible to send another command to cancel the `(check-sat)` request? The use-case is a user working in an IDE, where `(check-sat)` requests may quickly become obsolete because the user has continued typing.

**Comment by NikolajBjorner** [comment, marked as answer]
It isn't: the command interpreter waits for check-sat to complete, but you can send a CTL-C (interrupt) signal.
